HIP/hip-1056.md

```diff
@@ -476,35 +536,41 @@ enum BlockHashAlgorithm {
 }
 ```
 
-<aside>
-🚨 **Open Detail:** How do we define the `first_transaction_consensus_time` of a block with no transactions? Can it be
-first event consensus time if no transactions? what about empty block with no events?
-</aside>
+> Timestamp on empty round: Currently, for rounds with events and transactions, the round timestamp is equal to the
+timestamp of the last transaction in the round. For empty rounds with no events, the round timestamp is equal to the
+timestamp of the last transaction in the previous round plus 1,000 nanoseconds, rounded up to the nearest 1,000 nanoseconds.
+If there is no previous round, then the round timestamp is the median of the judge created times.
+This logic derives the value of the `BlockHeader` item `first_transaction_consensus_time` property. 
+
 
 #### EventHeader
 
 The event header depicts the beginning of an event. All items after this and before the next `EventHeader` or
 `BlockProof` are part of this event or results of execution of this event’s contents. It is designed so that it is easy
-to reconstruct `GossipEvent`’s as needed for hashgraph reconstruction from the contents in the block stream. To make
+to reconstruct `GossipEvent`s as needed for hashgraph reconstruction from the contents in the block stream. To make
 that easy `EventHeader` includes `EventCore` that is part of `GossipEvent` and all transactions are defined as
 `EventTransaction` which can be used unchanged to construct the original `GossipEvent`.
 
+> Note: If there are non-transaction state changes (i.e. platform state changes or the sum of all changes to the
+transaction receipt queue), those are not a result of the last event's execution or contents. 
+They are a consequence of the round as a whole.
```

Comment:
Sorry, I didn't mean for my words to be added verbatim. Platform state changes are made as a result of the round and not transactions. The sum of all changes to the transaction receipt queue are definitely from transactions. The unique aspect is that it is a single state change from many transactions.